implementations/c/ockam/channel/channel.h

```diff
@@ -0,0 +1,28 @@
+#ifndef OCKAM_CHANNEL_H
+#define OCKAM_CHANNEL_H
+
+#include "ockam/error.h"
+#include "ockam/io.h"
+#include "ockam/memory.h"
+#include "ockam/vault.h"
+
+#define CHANNEL_ERROR_PARAMS          (OCKAM_ERROR_INTERFACE_CHANNEL | 0x0001u)
```

Comment:
Consider using separate enum type for errors from specific module, this is both more readable (it's easier to find what the error is knowing it's type and code, rather that having some big integer and computing what's the error type bit & error code are) and eliminates need to have different non-repeatable error bit for each module (this involves excessive modules coupling)

Reply:
Agree 💯  `ockam/error.h` is a source of a lot of coupling.
I'm not sure if we even need the `ockam_error_t` as a type.

Could you prototype with a quick example in one module what this may look like.